mantidproject/mantid
# Set values in SlitCalculator based on data from Instrument Definition

Fixes #13758 

Previously, the values for the distance between the first two slits and the distance between the second slit and the sample holder were hard coded into the SlitCalculator widget for Reflectometry GUI. Now the slitCalculator should grab it's data from the definition of the Process/Search instrument that has been set in the gui.

**To Test**

ISIS Reflectometry:
- Open ISIS Reflectometry (`interfaces > Reflectometry > ISIS Reflectometry`)
- Go to Help Menu on the Option bar
- Open Slit Calculator (take note of the values in  `Slit1 -> Slit2` and `Slit2 -> SA`)
- Close Slit Calculator
- Change the instrument either in the `Search Runs` tab or to the right of the progress bar at the bottom
- Reopen Slit Calculator
- Observe that the values for `Slit1 -> Slit2` and `Slit2 -> SA` in the slit calculator change.
- Observe that the values for `Slit1` and `Slit2` are still being calculated in the output section of the widget.

Repeat change of instrument until all instruments in the drop-down menu have been chosen and the functionality has been observed for each.

POLREF:
- Open Reflectometry Polref (`interfaces > Reflectometry > ISIS Reflectometry (Polref)`)
- Go to Reflectomery Menu on the Option bar
- Open Slit Calculator (keep this open and in view while performing the next step)
- Change the instrument either in the `Search Runs` tab or to the right of the progress bar at the bottom
- Observe that the values for `Slit1 -> Slit2` and `Slit2 -> SA` in the slit calculator change.
- Observe that the values for `Slit1` and `Slit2` are still being calculated in the output section of the widget.

Repeat change of instrument until all instruments in the drop-down menu have been chosen and the functionality has been observed for each.

**Release Notes**
TODO
